### PR TITLE
updated graph.py

### PR DESCRIPTION
**Pull request by Virat-Shrimali**
Completed the task mentioned in the issue #22

### CODE REVIEW

File: package/graph.py [hunk 1]
```diff
@@ -1,38 +1,69 @@
...
+def copy_to_clipboard(text):
+    """
+    Copies the provided text to the clipboard.
+    
+    Args:
+        text (str): The text to copy to the clipboard.
+
+    Raises:
+        OSError: If the operating system is unsupported.
+    """
```
Comment (this-is-yaash) [inline package/graph.py:16]: remove these lines!

File: package/graph.py [hunk 2]
```diff
@@ -1,38 +1,69 @@
...
+        )
+    # Windows
+    elif "win32" in sys.platform:
+        subprocess.run(
+            ["C:\\Windows\\System32\\clip.exe"],
+            input=text.strip().encode(),
+            check=True,
+        )
+    else:
+        raise OSError("Unsupported operating system")
```
Comment (this-is-yaash) [inline package/graph.py:32]: Add these following code snippet as well.
```suggestion
        # macOS
        elif "darwin" in sys.platform:
            subprocess.run(
                ["/usr/bin/pbcopy"],  # Full path to pbcopy for macOS
                input=text.strip().encode(),
                check=True,
            )
        else:
            raise OSError("Unsupported operating system")
```

**Review by this-is-yaash — COMMENTED** [review]
Please make the necessary updates.

File: package/graph.py (hunk 2 above)
Comment (Virat-Shrimali) [reply package/graph.py:30]: Done

**Comment by Wahid7852** [comment]
Hey Virat! May I know why are you changing the order of the functions?

**Comment by Virat-Shrimali** [comment]
Typically, helper functions (like copy_to_clipboard) come before the main function (display), so that the main function can call any helper functions that have already been defined.

However, if you prefer to maintain the original order of the functions or if there’s a specific reason in your project to keep the original order, you can absolutely do that without any issues. The functionality remains the same regardless of the order of the function definitions, as long as they are defined before they are used.

**Comment by Wahid7852** [comment]
I believe to not touch the order of the functions for no reason
it makes useless additions to the number of lines changed and has no benefits.

It'd be really appreciated if you maintain the original order and just do the actual changes

**Review by Wahid7852 — APPROVED** [review]
LGTM

**Comment by Wahid7852** [comment]
@jindalpriyanshu101 @this-is-yaash review please

File: package/graph.py [hunk 3]
```diff
@@ -1,38 +1,53 @@
...
     )
 
-    with open(snippet_path, "r") as file:
-        source_code = file.read()
+    try:
+        with open(snippet_path, "r") as file:
+            source_code = file.read()
+
+        # Copy the source code to clipboard
+        copy_to_clipboard(source_code)
```
Comment (this-is-yaash) [inline package/graph.py:24]: not necessary to call the function inside a method.
remove these lines

**Review by this-is-yaash — COMMENTED** [review]
Just remove the line and you're good to go!

**Pull request merged**